src/main/java/org/sonatype/nexus/repository/composer/internal/ComposerJsonProcessor.java

```diff
@@ -239,6 +243,94 @@ public Content buildProviderJson(final Repository repository,
+    Map<String, Map<String, Object>> packages = new LinkedHashMap<>();
+    for (Payload payload : payloads) {
+      Map<String, Object> json = parseJson(payload);
+      if (json.get(PACKAGES_KEY) instanceof Map) {
+
+        Map<String, Object> packagesMap = (Map<String, Object>) json.get(PACKAGES_KEY);
+        for (String packageName : packagesMap.keySet()) {
+
+          Map<String, Object> packageVersions = (Map<String, Object>) packagesMap.get(packageName);
+          for (String packageVersion : packageVersions.keySet()) {
+
+            Map<String, Object> versionInfo = (Map<String, Object>) packageVersions.get(packageVersion);
+            Map<String, Object> distInfo = (Map<String, Object>) versionInfo.get(DIST_KEY);
+            if (distInfo == null) {
+              continue;
+            }
+
+            Map<String, Object> newDistInfo = new LinkedHashMap<>();
+            newDistInfo.put(URL_KEY, String
+                .format(REWRITE_URL, repository.getUrl(), packageName, packageVersion, packageName.replace('/', '-'),
+                    packageVersion));
+            newDistInfo.put(TYPE_KEY, distInfo.get(TYPE_KEY));
+
+            Map<String, Object> newPackageInfo = new LinkedHashMap<>();
+            newPackageInfo.put(NAME_KEY, versionInfo.get(NAME_KEY));
+            newPackageInfo.put(VERSION_KEY, versionInfo.get(VERSION_KEY));
+            newPackageInfo.put(DIST_KEY, newDistInfo);
+            newPackageInfo.put(TIME_KEY, time);
+            newPackageInfo.put(UID_KEY, Integer.toUnsignedLong(
+                Hashing.md5().newHasher()
+                    .putString(packageName, StandardCharsets.UTF_8)
+                    .putString(packageVersion, StandardCharsets.UTF_8)
+                    .putString(time, StandardCharsets.UTF_8)
+                    .hash()
+                    .asInt()));
+
+            if (versionInfo.containsKey(AUTOLOAD_KEY)) {
+              newPackageInfo.put(AUTOLOAD_KEY, versionInfo.get(AUTOLOAD_KEY));
+            }
+            if (versionInfo.containsKey(REQUIRE_KEY)) {
+              newPackageInfo.put(REQUIRE_KEY, versionInfo.get(REQUIRE_KEY));
+            }
+            if (versionInfo.containsKey(REQUIRE_DEV_KEY)) {
+              newPackageInfo.put(REQUIRE_DEV_KEY, versionInfo.get(REQUIRE_DEV_KEY));
+            }
+            if (versionInfo.containsKey(SUGGEST_KEY)) {
+              newPackageInfo.put(SUGGEST_KEY, versionInfo.get(SUGGEST_KEY));
+            }
+
+            if (!packages.containsKey(packageName)) {
+              packages.put(packageName, new LinkedHashMap<>());
+            }
+
+            Map<String, Object> packagesForName = packages.get(packageName);
+            packagesForName.putIfAbsent(packageVersion, newPackageInfo);
+          }
+        }
+      }
+    }
+
+    return new Content(new StringPayload(mapper.writeValueAsString(Collections.singletonMap(PACKAGES_KEY, packages)),
```

Comment:
Trivial: Collections doesn't provide any additional information here you could safely statically import it (i.e. In fact a map isn't strictly a "Collection" ... I wonder why they put the method there hmm).

Reply:
> Trivial: Collections doesn't provide any additional information here you could safely statically import it (i.e. In fact a map isn't strictly a "Collection" ... I wonder why they put the method there hmm).

Done in https://github.com/sonatype-nexus-community/nexus-repository-composer/pull/14/commits/70805ad368f979c55a8f9fcc5224057700d970d2.